src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2266,6 +2267,14 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
                         }
 
                         Span<byte> initCode = vmState.Memory.LoadSpan(in memoryPositionOfInitCode, initCodeLength);
+                        
+                        // Reject code starting with 0xEF if EIP-3541 is enabled.
+                        if (spec.IsEip3541Enabled && initCode.Length >= 1 && initCode[0] == 0xEF)
```

Comment:
Wait -- it looks like you're checking the `initcode` here -- you should be checking the to-be-deployed code

Reply:
The 3541 check can be implemented the same was as the deploy-code-size is checked, wherever that is. The semantics are the same as that one

Reply:
@MarekM25 
![image](https://user-images.githubusercontent.com/498913/118559438-d51a9f80-b75f-11eb-86db-dc1bc0000ea1.png)